src/main.c

```diff
@@ -302,7 +286,24 @@ main (int argc, char *argv[])
                               time_now, time_str_appended, current_waste_num);
 
             if (info_status == 0)
+            {
```

Comment:
Yes, please do another pull request and let's have a look. :)

I haven't examined your patch yet fully, but I suspect that it will need to be integrated into purge_rmw.c as well.